### ISSUE TITLE
Sometimes Null Pointer Exception is thrown while accessing Amplify.currentuser?, note: even if ? is used

### ISSUE THREAD

**Issue by SomnathS09** [aws-amplify/amplify-android#1489]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin, Kotlin - Coroutines

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line
// Amplify core dependency
    implementation 'com.amplifyframework:core:1.18.0'


    implementation 'com.amplifyframework:aws-auth-cognito:1.18.0'
    implementation 'com.amplifyframework:aws-storage-s3:1.18.0'


```

</details>

### Environment information

<details>

```
# Put output below this line
------------------------------------------------------------
Gradle 6.5
------------------------------------------------------------

Build time:   2020-06-02 20:46:21 UTC
Revision:     a27f41e4ae5e8a41ab9b19f8dd6d86d7b384dad4

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.10 (JetBrains s.r.o 11.0.10+0-b96-7249189)
OS:           Windows 10 10.0 amd64



```

</details>

### Please include any relevant guides or documentation you're referencing

https://github.com/aws-amplify/amplify-android/issues/851

### Describe the bug

Sometimes Null Pointer Exception is thrown while accessing Amplify.currentuser?, note: even if ? is used

`        Amplify.currentUser?.let {
Log.d("CurrentUser", $it )
        }`

See in the above code, I've specifically checked for null check and used let to avoid null cases of currentUser (obviously it will be null if no user is logged in)
But, even after using ? (elvis) and let, I am still getting null pointer exception internally in Amplify

Exception thrown:
java.lang.NullPointerException
        at java.util.Objects.requireNonNull(Objects.java:220)
        at com.amplifyframework.auth.AuthUser.<init>(AuthUser.java:37)
        at com.amplifyframework.auth.cognito.AWSCognitoAuthPlugin.getCurrentUser(AWSCognitoAuthPlugin.java:947)
        at com.amplifyframework.auth.AuthCategory.getCurrentUser(AuthCategory.java:297)
        at com.google.samples.apps.awslogin.LoginFragment.onCreateView(LoginFragment.kt:93)
        

### Reproduction steps (if applicable)

1. Try Logging in and Logging out multiple times, it happens sporadically 

### Code Snippet

```kotlin
// Put your code below this line.
Amplify.currentUser?.let {
Log.d("CurrentUser", $it )
        }
```

### Log output

<details>

```
// Put your logs below this line
java.lang.NullPointerException
        at java.util.Objects.requireNonNull(Objects.java:220)
        at com.amplifyframework.auth.AuthUser.<init>(AuthUser.java:37)
        at com.amplifyframework.auth.cognito.AWSCognitoAuthPlugin.getCurrentUser(AWSCognitoAuthPlugin.java:947)
        at com.amplifyframework.auth.AuthCategory.getCurrentUser(AuthCategory.java:297)
        at com.google.samples.apps.awslogin.LoginFragment.onCreateView(LoginFragment.kt:93)

```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by raphkim** [comment]
Thank you for reporting this @SomnathS09! This looks like a bug where Auth plugin assumes that `username` is guaranteed to be not null as long as `userId` is not null. I will investigate this to determine expected behavior and fix it accordingly.

**Comment by raphkim** [comment]
This should be resolved in our latest version 1.26.0!
